includes/woopay/class-woopay-adapted-extensions.php

```
@@ -0,0 +1,118 @@
+<?php
+/**
+ * WooPay
+ *
+ * @package WCPay\WooPay
+ */
+
+namespace WCPay\WooPay;
+
+use Automattic\WooCommerce\Blocks\Integrations\IntegrationInterface;
+
+/**
+ * WooPay
+ */
+class WooPay_Adapted_Extensions {
+	const POINTS_AND_REWARDS_PLUGIN = 'woocommerce-points-and-rewards';
+	const POINTS_AND_REWARDS_API    = 'points-and-rewards';
+
+	/**
+	 * WC Blocks registered integrations, as `$name => $instance` pairs.
+	 *
+	 * @var IntegrationInterface[]
+	 */
+	protected $registered_integrations = [];
+
+	/**
+	 * Initializa WC Blocks regitered integrations.
+	 */
+	public function __construct() {
+		do_action( 'woocommerce_blocks_checkout_block_registration', $this );
+	}
+
+	/**
+	 * Get WooPay adapted extensions settings and extra data needed on WooPay.
+	 *
+	 * @param string $email The user email the data will be loaded.
+	 */
+	public function get_adapted_extensions_data( $email ) {
+		$enabled_adapted_extensions = get_option( WooPay_Scheduler::ENABLED_ADAPTED_EXTENSIONS_OPTION_NAME, [] );
+		$extension_settings         = [];
+
+		if ( count( $enabled_adapted_extensions ) === 0 ) {
+			return null;
+		}
+
+		$user = wp_get_current_user();
+
+		if ( ! is_user_logged_in() ) {
+			// If the user is guest and has a merchant account, load data from there to know if need
+			// to verify their email on WooPay.
+			$user_by_email = get_user_by( 'email', $email );
+
+			if ( false !== $user_by_email ) {
+				$user = $user_by_email;
+			}
+		}
+
+		// Points and Rewards.
+		if ( in_array( self::POINTS_AND_REWARDS_PLUGIN, $enabled_adapted_extensions, true ) ) {
+			$extension_settings[ self::POINTS_AND_REWARDS_API ] = self::get_points_and_rewards_data( $user );
+		}
+
+		return $extension_settings;
+	}
+
+	/**
+	 * Get Points and Rewards settings for WooPay.
+	 *
+	 * @psalm-suppress UndefinedClass
```

Shouldn't the error be suppressed only in the line it happens? Suppressing it for the whole function could make it go unnoticed in case someone adds another Undefined Class in the future by mistake.

---

Fixed on https://github.com/Automattic/woocommerce-payments/pull/6637/commits/e14ff86eaf0f5da634d4dc86632ce77501793ea5.